src/osemgrep/cli/CLI.ml

```diff
@@ -31,12 +31,74 @@ open Common
    Translated from cli.py and commands/wrapper.py
 *)
 
+module Env = Semgrep_envvars
+
 (*****************************************************************************)
 (* Constants *)
 (*****************************************************************************)
 
 let default_subcommand = "scan"
 
+(*****************************************************************************)
+(* Helpers *)
+(*****************************************************************************)
+
+(* Placeholder for adaptation of pysemgrep state.terminal.init_for_cli() *)
+(* TOPORT:
+     1. GITHUB_ACTIONS specific output requirements
+     2. Any NO_COLOR / SEMGREP_FORCE_NO_COLOR behavior
+*)
+(* let init_for_cli () : unit =
+   ()
+*)
+
+let metrics_init () : unit =
+  let () =
+    match Env.v.user_agent_append with
+    | Some str -> Metrics_.add_user_agent_tag ~str
+    | _ -> ()
+  in
+  let settings = Semgrep_settings.load () in
+  let api_token = settings.Semgrep_settings.api_token in
+  let anonymous_user_id = settings.Semgrep_settings.anonymous_user_id in
+  Metrics_.init ~anonymous_user_id ~ci:Env.v.is_ci;
```

Comment:
seems weird to have a call to Metrics_.add_user_agent_tag before a call to Metrics_.init

Reply:
yeah – I'll add a comment.